service/src/main/java/bio/terra/pipelines/service/JobsService.java

```diff
@@ -83,48 +91,33 @@ protected UUID writeJobToDb(
     job.setTimeCompleted(null);
     job.setStatus(status);
 
-    if (attempt <= 3) {
-      Job createdJob = writeJobToDbRetryDuplicateException(job);
-      if (createdJob == null) {
-        int nextAttempt = attempt + 1;
-        return writeJobToDb(
-            userId,
-            pipelineId,
-            pipelineVersion,
-            timeSubmitted,
-            status,
-            pipelineInputs,
-            nextAttempt);
-      } else {
-        // once job is created save related pipeline inputs
-        PipelineInput pipelineInput = new PipelineInput();
-        pipelineInput.setJobId(createdJob.getId());
-        pipelineInput.setInputs(pipelineInputs.toString());
-        pipelineInputsRepository.save(pipelineInput);
-        return createdJob.getJobId();
-      }
-    } else {
-      // 3 attempts to write a job to the database failed
-      return null;
-    }
+    Job createdJob = writeJobToDbThrowsDuplicateException(job);
+
+    // once job is created save related pipeline inputs
+    PipelineInput pipelineInput = new PipelineInput();
+    pipelineInput.setJobId(createdJob.getId());
+    pipelineInput.setInputs(pipelineInputs.toString());
+    pipelineInputsRepository.save(pipelineInput);
+
+    return createdJob.getJobId();
   }
 
-  protected Job writeJobToDbRetryDuplicateException(Job job) {
+  protected Job writeJobToDbThrowsDuplicateException(Job job) throws DuplicateObjectException {
     try {
       jobsRepository.save(job);
       logger.info("job saved for jobId: {}", job.getJobId());
     } catch (DataIntegrityViolationException e) {
       if (e.getCause() instanceof ConstraintViolationException) {
-        logger.warn("Duplicate jobId {} found, retrying", job.getJobId());
-        return null;
+        throw new DuplicateObjectException(
+            String.format("Duplicate jobId %s found", job.getJobId()));
       }
       throw e;
     }
 
     return job;
   }
 
-  private Instant getCurrentTimestamp() {
+  public Instant getCurrentTimestamp() {
```

Comment:
hmm i think we can use postgres itself to set these timestamps for us when rows are created in the database.  This is a mysql example but something like [this](https://stackoverflow.com/questions/19230489/default-value-on-update-liquibase)

Reply:
great call. made TSPS-129 to do this